redis/tests/test_basic.rs

```diff
@@ -1448,3 +1449,32 @@ fn test_blocking_sorted_set_api() {
         );
     }
 }
+
+#[test]
+fn test_set_client_name_by_config() {
+    const CLIENT_NAME: &str = "TEST_CLIENT_NAME";
+
+    let ctx = TestContext::with_clientname(CLIENT_NAME);
+    let mut con = ctx.connection();
```

Comment:
notice that you're testing the sync connection, which is the connection we care less about.

Reply:
Yep, I am absolutely aware. The intent is to test the API and the configuration propagation , not the method of invocation. I consider it the same value as the basic get/set tests

Reply:
need to add to async also